paper_figures/code/maxsavings_contour.py

```diff
@@ -30,31 +33,45 @@
 mef_data = getmef(region, month)
 aef_data = getaef(region, month)
 dam_data = getdam(region, month)
+tariff_data = gettariff(region)
 
 # solve max mef savings in parallel for a range of uptime and continuous flex
 uptimes = np.arange(0, 25, 2) / 24  # 1 to 24 hours to percent in intervals
 continuous_flex = np.arange(0, 1.01, 0.1)  # 0 to 100%
 
-baseload = np.ones_like(mef_data)  # 1MW flat baseline load
+non_tariff_baseload = np.ones_like(mef_data)  # 1MW flat baseline load
+
+# tariffs require a full month of data to properly incorporate monthly demand charges
+startdate_dt, enddate_dt = get_start_end(month)
+month_length = int((enddate_dt - startdate_dt) / np.timedelta64(1, "h"))
+tariff_baseload = np.ones(month_length)
 
 # create a container for results
 max_mef_savings_results = np.zeros((len(uptimes), len(continuous_flex)))
 max_aef_savings_results = np.zeros((len(uptimes), len(continuous_flex)))
 max_dam_savings_results = np.zeros((len(uptimes), len(continuous_flex)))
+max_tariff_savings_results = np.zeros((len(uptimes), len(continuous_flex)))
 
 # calculate max mef savings
 for i, uptime in enumerate(uptimes):
     for j, flex in enumerate(continuous_flex):
         max_mef_savings_results[i, j] = max_mef_savings(
-            mef_data, uptime, flex, baseload
+            mef_data, uptime, flex, non_tariff_baseload
         )
         max_aef_savings_results[i, j] = max_aef_savings(
-            aef_data, uptime, flex, baseload
+            aef_data, uptime, flex, non_tariff_baseload
         )
         max_dam_savings_results[i, j] = max_dam_savings(
-            dam_data, uptime, flex, baseload
+            dam_data, uptime, flex, non_tariff_baseload
+        )
+        max_tariff_savings_results[i, j] = max_tar_savings(
```

Comment:
Following prev. comment in `maxsavings.py` - tariff optimization for contour should have uptime equality constraint = True.